qr_generator/lib/screens/scan_view/scan_view.dart

```diff
@@ -0,0 +1,162 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:permission_handler/permission_handler.dart';
+import 'package:qr_generator/screens/scan_view/bloc/scan_view_bloc.dart';
+import 'package:qr_generator/screens/widgets/error_view.dart';
+import 'dart:io';
+import 'package:flutter/foundation.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+
+class ScanView extends StatefulWidget {
+  ScanView({Key key}) : super(key: key);
+
+  @override
+  _ScanViewState createState() => _ScanViewState();
+}
+
+class _ScanViewState extends State<ScanView> {
+  ThemeData _theme;
+  ScanViewBloc _scanViewBloc;
+  Barcode result;
+  QRViewController controller;
+  final GlobalKey qrKey = GlobalKey(debugLabel: 'QR');
+
+  @override
+  void initState() {
+    super.initState();
+    _scanViewBloc = BlocProvider.of<ScanViewBloc>(context);
```

Comment:
**question:** why store this as a var in this stateful widget instead of pulling it like you're doing here where it's needed?

Reply:
Done!

Reply:
I normally use it this way because I believe it's cleaner this way. But, I agree with your suggestion and changed it.